algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp

```diff
@@ -0,0 +1,134 @@
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <TestStdAlgorithmsCommon.hpp>
+#include <utility>
+#include <Kokkos_Random.hpp>
+
+namespace Test {
+namespace stdalgos {
+namespace MoveBackward {
+
+namespace KE = Kokkos::Experimental;
+
+template <class Tag, class ValueType, class InfoType>
+void run_single_scenario(const InfoType& scenario_info, int apiId) {
+  const std::size_t view_ext = std::get<1>(scenario_info);
+
+  auto v = create_view<ValueType>(Tag{}, view_ext, "v");
+
+  // v might not deep copyable so to modify it on the host
+  // need to do this
+  auto v_dc   = create_deep_copyable_compatible_view_with_same_extent(v);
+  auto v_dc_h = create_mirror_view(Kokkos::HostSpace(), v_dc);
+  Kokkos::Random_XorShift64_Pool<Kokkos::DefaultHostExecutionSpace> pool(12371);
+  Kokkos::fill_random(v_dc_h, pool, 0, 523);
+  // copy to v_dc and then to v
+  Kokkos::deep_copy(v_dc, v_dc_h);
+  CopyFunctor<decltype(v_dc), decltype(v)> F1(v_dc, v);
+  Kokkos::parallel_for("copy", v.extent(0), F1);
+
+  // make a gold copy of v right now because we call the algorithm
+  auto gold = create_host_space_copy(v);
+
+  // create another view that is bigger than v
+  // because we need it to test the move_backward
+  auto v2 = create_view<ValueType>(Tag{}, view_ext + 5, "v2");
+
+  if (apiId == 0) {
+    auto rit =
+        KE::move_backward(exespace(), KE::begin(v), KE::end(v), KE::end(v2));
+    const int dist = KE::distance(KE::begin(v2), rit);
+    EXPECT_TRUE(dist == 5);
+  } else if (apiId == 1) {
+    auto rit       = KE::move_backward("mylabel", exespace(), KE::begin(v),
+                                 KE::end(v), KE::end(v2));
+    const int dist = KE::distance(KE::begin(v2), rit);
+    EXPECT_TRUE(dist == 5);
+  } else if (apiId == 2) {
+    auto rit       = KE::move_backward(exespace(), v, v2);
+    const int dist = KE::distance(KE::begin(v2), rit);
+    EXPECT_TRUE(dist == 5);
+  } else if (apiId == 2) {
```

Comment:
`apiId == 2` is already covered in line 89
```suggestion
  } else if (apiId == 3) {
```

Reply:
good catch